src/main/java/emissary/util/GitRepositoryState.java

```diff
@@ -0,0 +1,147 @@
+package emissary.util;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GitRepositoryState {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GitRepositoryState.class);
+    private static GitRepositoryState gitRepositoryState = null;
+
+    private final String tags;
+    private final String branch;
+    private final String dirty;
+    private final String remoteOriginUrl;
+    private final String commitIdAbbrev;
+    private final String describe;
+    private final String describeShort;
+    private final String commitUserName;
+    private final String commitUserEmail;
+    private final String commitMessageFull;
+    private final String commitMessageShort;
+    private final String commitTime;
+    private final String closestTagName;
+    private final String closestTagCommitCount;
+    private final String buildUserName;
+    private final String buildUserEmail;
+    private final String buildTime;
+    private final String buildHost;
+    private final String buildVersion;
+
+    private GitRepositoryState(Properties properties) {
+
+        this.tags = properties.get("git.tags").toString();
+        this.branch = properties.get("git.branch").toString();
+        this.dirty = properties.get("git.dirty").toString();
+        this.remoteOriginUrl = properties.get("git.remote.origin.url").toString();
+
+        this.commitIdAbbrev = properties.get("git.commit.id.abbrev").toString();
+        this.describe = properties.get("git.commit.id.describe").toString();
+        this.describeShort = properties.get("git.commit.id.describe-short").toString();
+        this.commitUserName = properties.get("git.commit.user.name").toString();
+        this.commitUserEmail = properties.get("git.commit.user.email").toString();
+        this.commitMessageFull = properties.get("git.commit.message.full").toString();
+        this.commitMessageShort = properties.get("git.commit.message.short").toString();
+        this.commitTime = properties.get("git.commit.time").toString();
+        this.closestTagName = properties.get("git.closest.tag.name").toString();
+        this.closestTagCommitCount = properties.get("git.closest.tag.commit.count").toString();
+
+        this.buildUserName = properties.get("git.build.user.name").toString();
+        this.buildUserEmail = properties.get("git.build.user.email").toString();
+        this.buildTime = properties.get("git.build.time").toString();
+        this.buildHost = properties.get("git.build.host").toString();
+        this.buildVersion = properties.get("git.build.version").toString();
+    }
+
+    public static GitRepositoryState getRepositoryState() {
```

Comment:
Maybe allow the properties file to be passed in.
```suggestion
    public static GitRepositoryState getRepositoryState(String gitProperties) {
```